core/database_api.php

```diff
@@ -838,12 +838,27 @@ function db_prepare_bool( $p_bool ) {
 }
 
 /**
- * return current timestamp for DB
- * @todo add param bool $p_gmt whether to use GMT or current timezone (default false)
- * @return string Formatted Date for DB insertion e.g. 1970-01-01 00:00:00 ready for database insertion
+ * return current time as Unix timestamp
+ * @param boolean $p_gmt True: current time is GMT; False: current time is in the current timezone (default)
+ * @return int Unix timestamp of the current date and time
  */
-function db_now() {
-	return time();
+function db_now( $p_gmt = false ) {
+    $t_now = time();
+    if( $p_gmt )
+        $t_now += (int)date('Z');
+	return $t_now;
+}
+
+/**
+ * return current time as DATETIME-formatted date and time
+ * @param boolean $p_gmt True: current time is GMT; False: current time is in the current timezone (default)
+ * @return string DATETIME-formatted date, such as "1970-01-01 00:00:00"
+ */
+function db_now_datetime( $p_gmt = false ) {
+    $t_now = time();
+    if( $p_gmt )
+        $t_now += (int)date('Z');
+	return date( 'Y-m-d H:i:s', $t_now);
```

Comment:
I wouldn't add the new function unless we have a clearly identified need for it.